src/App.js

```diff
@@ -1,21 +1,176 @@
+    super(props)
+    this.state = {
+      user: null,
+      userLocation: LANDING,
+      firstName: '',
+      lastName: '',
+      email: '',
+      phone: '',
+      password: '',
+      avatar: ''
+    }
+    this.toLogIn = this.toLogIn.bind(this)
+    this.toSignUp = this.toSignUp.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+    this.logIn = this.logIn.bind(this)
+    this.setUser = this.setUser.bind(this)
+    this.clearUser = this.clearUser.bind(this)
+  }
+
+  toLogIn () {
+    this.setState({userLocation: LOG_IN})
+  }
+
+  logIn (e) {
+    e.preventDefault()
+    let userUID
+    const setUser = this.setUser
+    firebase.auth().signInWithEmailAndPassword(this.state.email, this.state.password)
+      .then(
+        firebase.auth().onAuthStateChanged(function (fbuser) {
+          if (fbuser) {
+            userUID = fbuser.uid
+            setUser(userUID)
+          } else {
+            userUID = null
+            setUser(userUID)
+          }
+        })
+      )
+      .catch(function (error) {
+        var errorCode = error.code
+        var errorMessage = error.message
+        console.log('catch ran on logIn', errorCode, errorMessage)
+      })
+  }
+
+  clearUser () {
+    this.setState({user: null})
+  }
+
+  setUser (userUID) {
+    let nextPage
+    if (userUID) {
+      nextPage = PROFILES
+    } else {
+      nextPage = LOG_IN
+    }
+    this.setState({user: userUID, userLocation: nextPage})
+  }
+
+  toSignUp () {
+    this.setState({userLocation: SIGN_UP})
+  }
+
+  handleChange (e) {
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+
+  handleSubmit (e) {
+    e.preventDefault()
+    firebase.auth().createUserWithEmailAndPassword(this.state.email, this.state.password)
+      .then((response) => {
+        const itemsRef = firebase.database().ref('users')
+        const item = {
+          firstName: this.state.firstName,
+          lastName: this.state.lastName,
+          email: this.state.email,
+          phone: this.state.phone,
+          avatar: this.state.avatar
+        }
+        const onComplete = function (error) {
+          if (error) {
+            console.log('Operation failed')
+          } else {
+            console.log(' Operation completed')
+          }
+        }
+        itemsRef.child(response.uid).set(item, onComplete)
+      })
+      .catch(function (error) {
+        var errorCode = error.code
+        var errorMessage = error.message
+        console.log('catch ran', errorCode, errorMessage)
+      })
+    let userUID
+    const setUser = this.setUser
+    firebase.auth().onAuthStateChanged(function (fbuser) {
+      if (fbuser) {
+        userUID = fbuser.uid
+        setUser(userUID)
+      } else {
+        userUID = null
+        setUser(userUID)
+      }
+    })
+    this.setState({userLocation: PROFILES})
+  }
+
+  render () {
+    if (this.state.userLocation === LANDING) {
```

Comment:
You're going to need to route in this app, so no time like the present. You can do this in a follow-up PR if you like. Since you're going to need routing anyway, this kind of pattern should not be used. Simply use routes to indicate your state. As we are likely to use Redux in the near future, I would recommend using `react-little-router`

Reply:
will be implemented in next PR